### DISCUSSION TITLE
Getting rows from `RecordBatch`

### DISCUSSION THREAD

**Discussion by kskalski** [apache/arrow-rs#4435] [Q&A] [answered]
I'm looking at the `async` API for reading parquet files and it offers creating stream of `RecordBatch`. Right now I'm using `parquet::file::serialized_reader::SerializedFileReader` for getting iterator over rows of the file, but I'm a bit lost on how to achieve similar access when using `async` and `RecordBatch`, which just gives raw access to columns.

Is this kind of conversion into `Row` objects easy to perform using existing APIs? Should row reader over `RecordBatch` be added as a new feature to `parquet::record` module? Or is there going to be `parquet::file::async_reader::AsyncFileReader`?

**Comment by Jefffrey** [comment, marked as answer]
There is an example here of iterating record batches in a row like fashion:

https://github.com/apache/arrow-rs/blob/12091af35e5070dae33e86fbe988d9319521cd16/arrow-avro/examples/decode_kafka_stream.rs#L122-L139

- Need to downcast the columns to expected types
- Using `value()` to get the actual value; note this should ideally be checked with `is_null()` prior

> Is this kind of conversion into `Row` objects easy to perform using existing APIs?

Its definitely a bit cumbersome

> Should row reader over `RecordBatch` be added as a new feature to `parquet::record` module? Or is there going to be `parquet::file::async_reader::AsyncFileReader`?

I'm not sure if we have appetite for such changes, but can always raise an issue for a feature request if desired
